### PR TITLE
fix: (IAC-1229) coalescelist failed: no non-null arguments error with byo_network_scenario=3

### PR DESCRIPTION
**Pull request by dhoucgitter**
# Changes
Terraform plan/apply error could occur with a byo_network_scenario=3 configuration when a security_group_id input variable value is set.
- only call the coalescelist function in the locals assignment when at least one of the security group tuple arguments is not empty.
- add BYOnetwork table required variables column content to align with "Subnet requirements" documentation paragraph, auto tf formatting, fix typo in ingress rule description.
# Tests
| BYON Scenario | tf plan | tf apply | Notes |
| -:|-|-|-|
| 0 | successful | skipped | |
| 2 | successful | successful | Successful EKS cluster create and destroy |
| 3 | reported error fixed, successful | reported error fixed, successful | Known issue with missing required security group attribute prevented EKS nodes from joining the cluster, not an IAC aws issue. **Update**: Successful EKS cluster created after adding existing SG ingress rule allowing nodes to join cluster. |

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -372,7 +372,7 @@ variable "subnets" {
   description = "Subnets to be created and their settings - This variable is ignored when `subnet_ids` is set (AKA bring your own subnets)."
   type        = map(list(string))
   default = {
-    "private" : ["192.168.0.0/18"], # multi-zonal cluster is created by adding additional subnets here
+    "private" : ["192.168.0.0/18"],                              # multi-zonal cluster is created by adding additional subnets here
```
Comment (jarpat) [inline variables.tf:375]: All the changes look good, may need a rebase the comment was removed in #241

Comment (dhoucgitter) [reply variables.tf:375]: Thanks @jarpat, it looks like that same comment was removed from CONFIG-VARS.md but remained in variables.tf, I'll remove it here as well.

File: locals.tf [hunk 2]
```diff
@@ -7,7 +7,7 @@ locals {
   aws_caller_identity_user_name = element(split("/", data.aws_caller_identity.terraform.arn), length(split("/", data.aws_caller_identity.terraform.arn)) - 1)
 
   # General
-  sec_group                 = coalescelist(aws_security_group.sg_a, aws_security_group.sg_b)
+  sec_group                 = (length(aws_security_group.sg_a) == 0 && length(aws_security_group.sg_b) == 0) ? null : coalescelist(aws_security_group.sg_a, aws_security_group.sg_b)
```
Comment (thpang) [inline locals.tf:10]: you'll run into issues here as length requires a length or a valid object if the object is null or can be set to null anywhere you'll need to do something like a can() on the items.

Comment (dhoucgitter) [reply locals.tf:10]: @thpang , for now aws_security_groups sg_a and sg_b are security group resources that will either have a valid non zero length if either one of them is defined or be an empty tuple of length zero when their count expressions evaluate to zero (can be true for both at once). Those objects are not being set anywhere else or to null at present so expect what I have is safe now for each of the tested scenarios. Does that mitigate the concern or need to use can() or try()? I looked at both and can consider using them if you recommend it.

**Review by thpang — APPROVED** [review]

**Review by jarpat — APPROVED** [review]

**Pull request merged**